src/app/highlight-by-amount.directive.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  inject,
+  input,
+  OnInit,
+  signal,
+} from '@angular/core';
+
+@Directive({
+  selector: '[appHighlightByAmount]',
+  standalone: true,
+})
+export class HighlightByAmount implements OnInit {
+  amount = input.required<number>({ alias: 'appHighlightByAmount' });
+  private elementRef = inject(ElementRef);
+  borderColor = signal('');
+
+  setDefaultColor() {
+    this.elementRef.nativeElement.style.color =
+      this.amount() > 0 ? '#45a049' : '#a04545';
+  }
+
+  ngOnInit() {
+    this.setDefaultColor();
+  }
+
+  updateBorderColor = () => {
+    if (this.amount() > 0) {
+      if (this.amount() < 500) {
+        this.borderColor.set('#5bff63');
+      } else if (this.amount() > 500 && this.amount() < 1000) {
+        this.borderColor.set('#45a049');
+      } else {
+        this.borderColor.set('#295f2c');
```

Comment:
If `this.amount() === 500` what should happen?

Reply:
I'll resolve this in the hover issue that you opened.